### PR TITLE
[AdvancedSearch] Fix PHPUnit 8 compatibility issues

### PR DESCRIPTION
**Pull request by dmytro-ch**
### Description (*)
This PR fixes the PHPUnit 8 compatibility issues of existing unit tests for `Magento_AdvancedSearch` module. Performed minor code refactoring and code style fixes.

### Related Pull Requests
N/A

### Fixed Issues (if relevant)

1. magento/magento2#27500: Unit Tests incompatible with PHPUnit 8

### Manual testing scenarios (*)

1. Run tests with PHPUnit 6: `vendor/bin/phpunit -c dev/tests/unit/phpunit.xml.dist app/code/Magento/AdvancedSearch/`
2. Install PHPUnit 8 and run the same tests.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/AdvancedSearch/Test/Unit/Model/Indexer/Fulltext/Plugin/CustomerGroupTest.php [hunk 1]
```diff
@@ -3,52 +3,65 @@
...
+    /**
+     * @var IndexerInterface|MockObject
      */
     protected $indexerMock;
 
     /**
-     * @var \PHPUnit_Framework_MockObject_MockObject|\Magento\Customer\Model\ResourceModel\Group
+     * @var Group|MockObject
      */
     protected $subjectMock;
```
Comment (rogyar) [inline app/code/Magento/AdvancedSearch/Test/Unit/Model/Indexer/Fulltext/Plugin/CustomerGroupTest.php]: How do you like the idea to change scope of the properties for this test case to `private`?

File: app/code/Magento/AdvancedSearch/Test/Unit/Controller/Adminhtml/Search/System/Config/TestConnectionTest.php [hunk 2]
```diff
@@ -3,70 +3,82 @@
...
-     * @var \Magento\Framework\Controller\Result\Json|\PHPUnit_Framework_MockObject_MockObject
+     * @var Json|MockObject
      */
     private $resultJson;
 
     /**
-     * @var \Magento\Framework\Controller\Result\JsonFactory|\PHPUnit_Framework_MockObject_MockObject
+     * @var JsonFactory|MockObject
      */
     private $resultJsonFactory;
```
Comment (lbajsarowicz) [inline app/code/Magento/AdvancedSearch/Test/Unit/Controller/Adminhtml/Search/System/Config/TestConnectionTest.php]: Add mock suffix to property name

File: app/code/Magento/AdvancedSearch/Test/Unit/Block/SearchDataTest.php [hunk 3]
```diff
@@ -3,58 +3,72 @@
...
+    /**
+     * @var TemplateContext|MockObject
+     */
     private $context;
 
     /**
-     * @var \Magento\Search\Model\QueryFactoryInterface|MockObject
+     * @var QueryFactoryInterface|MockObject
      */
     private $queryFactory;
```
Comment (lbajsarowicz) [inline app/code/Magento/AdvancedSearch/Test/Unit/Block/SearchDataTest.php]: Add mock to property name

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
The only thing needs to be verified that if property is mock, the property name should have 'Mock' suffix

File: app/code/Magento/AdvancedSearch/Test/Unit/Controller/Adminhtml/Search/System/Config/TestConnectionTest.php [hunk 4]
```diff
@@ -3,70 +3,82 @@
...
+use Magento\Framework\Controller\Result\JsonFactory;
+use Magento\Framework\Filter\StripTags;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
 
 /**
- * Class TestConnectionTest
- *
+ * @covers TestConnection
```
Comment (rogyar) [inline app/code/Magento/AdvancedSearch/Test/Unit/Controller/Adminhtml/Search/System/Config/TestConnectionTest.php]: The imported alias might not be recognized well by static tests. You may use FQN for the `@covers` annotation.

File: app/code/Magento/AdvancedSearch/Test/Unit/Model/Indexer/Fulltext/Plugin/CustomerGroupTest.php (hunk 1 above)
Comment (dmytro-ch) [reply app/code/Magento/AdvancedSearch/Test/Unit/Model/Indexer/Fulltext/Plugin/CustomerGroupTest.php]: Indeed. Thank you for noticing!

File: app/code/Magento/AdvancedSearch/Test/Unit/Controller/Adminhtml/Search/System/Config/TestConnectionTest.php (hunk 4 above)
Comment (dmytro-ch) [reply app/code/Magento/AdvancedSearch/Test/Unit/Controller/Adminhtml/Search/System/Config/TestConnectionTest.php]: Sure. I've fixed this typo :)

File: app/code/Magento/AdvancedSearch/Test/Unit/Controller/Adminhtml/Search/System/Config/TestConnectionTest.php (hunk 2 above)
Comment (dmytro-ch) [reply app/code/Magento/AdvancedSearch/Test/Unit/Controller/Adminhtml/Search/System/Config/TestConnectionTest.php]: Totally agree. Thank you!

**Review by lbajsarowicz — APPROVED** [review]
:heavy_check_mark: Thank you for your contribution.

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-7261](https://jira.corp.magento.com/browse/ENGCOM-7261) has been created to process this Pull Request
:eight_spoked_asterisk: @lbajsarowicz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
QA not applicable

**Pull request merged**